gms/gossiper.cc

```diff
@@ -1065,7 +1065,8 @@ future<> gossiper::replicate(inet_address ep, application_state key, const versi
 }
 
 future<> gossiper::advertise_removing(inet_address endpoint, utils::UUID host_id, utils::UUID local_host_id, abort_source& as) {
-    return seastar::async([this, g = this->shared_from_this(), endpoint, host_id, local_host_id, &as] {
+  return seastar::with_gate(_gate, [this, g = this->shared_from_this(), endpoint, host_id, local_host_id, &as] {
```

Comment:
class gossiper inherits from class seastar::async_sharded_service. This guarantees sharded::stop() will wait until all the references disappear before returning. This looks redundant to me. In general I like using gate to wait for the existing and prevent further operations, but it is not enough just for those functions you changed here. 

So, I would recommend to contain only the abort source changes in this PR so we can get it merged faster.

Reply:
OK, I dropped the last two commits (related to the gate).